In [2]:
import pandas as pd
pd.set_option("display.max_rows", 100, "display.max_columns", 100)

In [3]:
PATH = 'D:/project/데이터 분석 프로젝트/이노포스트/data/04_목욕장/'

df = pd.read_csv(PATH + '목욕장업.csv', encoding='cp949')

c:\Users\JBY\anaconda3\envs\3.7\lib\site-packages\IPython\core\interactiveshell.py:3524: DtypeWarning: Columns (40) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17190 entries, 0 to 17189
Data columns (total 51 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   번호           17190 non-null  int64  
 1   개방서비스명       17190 non-null  object 
 2   개방서비스아이디     17190 non-null  object 
 3   개방자치단체코드     17190 non-null  int64  
 4   관리번호         17190 non-null  object 
 5   인허가일자        17190 non-null  int64  
 6   인허가취소일자      0 non-null      float64
 7   영업상태구분코드     17190 non-null  int64  
 8   영업상태명        17190 non-null  object 
 9   상세영업상태코드     17190 non-null  int64  
 10  상세영업상태명      17190 non-null  object 
 11  폐업일자         10874 non-null  float64
 12  휴업시작일자       0 non-null      float64
 13  휴업종료일자       0 non-null      float64
 14  재개업일자        0 non-null      float64
 15  소재지전화        15920 non-null  object 
 16  소재지면적        17146 non-null  object 
 17  소재지우편번호      16708 non-null  float64
 18  소재지전체주소      17169 non-null  object 
 19  도로명전

In [5]:
df.head()

번호 개방서비스명    개방서비스아이디  개방자치단체코드                    관리번호     인허가일자  인허가취소일자  \
0   1   목욕장업  11_44_01_P   5220000  5220000-202-2012-00005  20120726      NaN   
1   2   목욕장업  11_44_01_P   5220000  5220000-202-2012-00006  20120824      NaN   
2   3   목욕장업  11_44_01_P   5220000  5220000-202-2012-00007  20121107      NaN   
3   4   목욕장업  11_44_01_P   5220000  5220000-202-1997-00003  19971115      NaN   
4   5   목욕장업  11_44_01_P   5220000  5220000-202-2003-00003  20030830      NaN   

   영업상태구분코드  영업상태명  상세영업상태코드 상세영업상태명  폐업일자  휴업시작일자  휴업종료일자  재개업일자  \
0         1  영업/정상         1      영업   NaN     NaN     NaN    NaN   
1         1  영업/정상         1      영업   NaN     NaN     NaN    NaN   
2         1  영업/정상         1      영업   NaN     NaN     NaN    NaN   
3         1  영업/정상         1      영업   NaN     NaN     NaN    NaN   
4         1  영업/정상         1      영업   NaN     NaN     NaN    NaN   

          소재지전화   소재지면적   소재지우편번호                   소재지전체주소  \
0  054 974 8383  349.46  718832.0   경상북도 칠곡군 석적읍 반계리 71-4번지   
1           NaN  489.79  718852.0   경상북도 칠곡군 기산면 행정리 47-2번지   
2  054 973 0888  540.97  718821.0     경상북도 칠곡군 지천면 송정리 45번지   
3   054 9749703  168.00  718813.0  경상북도 칠곡군 약목면 복성리 859-1번지   
4  054 975 9027  816.76  718811.0   경상북도 칠곡군 약목면 남계리 73-1번지   

                   도로명전체주소  도로명우편번호        사업장명          최종수정시점 데이터갱신구분  \
0   경상북도 칠곡군 석적읍 호국로 582-2  39848.0    반계숯가마찜질방  20120726132906       I   
1    경상북도 칠곡군 기산면 선노로 1082  39915.0  무시로장수마을찜질방  20161025095418       I   
2  경상북도 칠곡군 지천면 송정도당길 67-9  39867.0    솔밭에온가마찜질  20191004101355       U   
3    경상북도 칠곡군 약목면 약목로 87-8  39825.0       복성목욕탕  20130402100917       I   
4      경상북도 칠곡군 약목면 약목로 49  39827.0       약목목욕탕  20190507132922       U   

                 데이터갱신일자      업태구분명        좌표정보(x)        좌표정보(y)      위생업태명  \
0  2018-08-31 23:59:59.0  찜질시설서비스영업  329518.461747  282201.682569  찜질시설서비스영업   
1  2018-08-31 23:59:59.0  찜질시설서비스영업  325101.041208  275060.260588  찜질시설서비스영업   
2  2019-10-06 02:40:00.0  찜질시설서비스영업  332456.755220  275605.443845  찜질시설서비스영업   
3  2018-08-31 23:59:59.0       공동탕업  322935.849074  282378.690977       공동탕업   
4  2019-05-13 02:40:00.0       공동탕업  323189.818600  282060.794671       공동탕업   

   건물지상층수  건물지하층수  사용시작지상층  사용끝지상층  사용시작지하층  사용끝지하층  한실수  양실수  욕실수 발한실여부  의자수  \
0     2.0     0.0      1.0     2.0      NaN     NaN  0.0  0.0  2.0     Y  0.0   
1     1.0     0.0      1.0     1.0      NaN     NaN  0.0  0.0  0.0     Y  0.0   
2     1.0     0.0      1.0     1.0      0.0     0.0  0.0  0.0  0.0     N  0.0   
3     1.0     NaN      1.0     1.0      NaN     NaN  NaN  NaN  2.0     Y  NaN   
4     3.0     1.0      1.0     3.0      1.0     1.0  NaN  NaN  2.0     N  NaN   

  조건부허가신고사유  조건부허가시작일자  조건부허가종료일자 건물소유구분명  세탁기수  여성종사자수  남성종사자수  회수건조수  침대수  \
0       NaN        NaN        NaN      자가   0.0     NaN     NaN    0.0  0.0   
1       NaN        NaN        NaN      임대   0.0     NaN     NaN    0.0  0.0   
2       NaN        NaN        NaN      자가   0.0     NaN     NaN    0.0  0.0   
3       NaN        NaN        NaN      임대   NaN     NaN     NaN    NaN  NaN   
4       NaN        NaN        NaN      임대   NaN     NaN     NaN    NaN  NaN   

  다중이용업소여부  Unnamed: 50  
0        N          NaN  
1        N          NaN  
2        N          NaN  
3        N          NaN  
4        N          NaN

In [6]:
df.isnull().sum()

번호                 0
개방서비스명             0
개방서비스아이디           0
개방자치단체코드           0
관리번호               0
인허가일자              0
인허가취소일자        17190
영업상태구분코드           0
영업상태명              0
상세영업상태코드           0
상세영업상태명            0
폐업일자            6316
휴업시작일자         17190
휴업종료일자         17190
재개업일자          17190
소재지전화           1270
소재지면적             44
소재지우편번호          482
소재지전체주소           21
도로명전체주소         7682
도로명우편번호         7960
사업장명               0
최종수정시점             0
데이터갱신구분            0
데이터갱신일자            0
업태구분명              0
좌표정보(x)         1208
좌표정보(y)         1208
위생업태명              0
건물지상층수          4272
건물지하층수          5458
사용시작지상층         5783
사용끝지상층          8587
사용시작지하층         7737
사용끝지하층         10661
한실수             7397
양실수             7396
욕실수             4729
발한실여부             79
의자수             7388
조건부허가신고사유      17163
조건부허가시작일자      17171
조건부허가종료일자      17170
건물소유구분명        12649
세탁기수           10571
여성종사자수         15390
남성종사자수         15390
회수건조수        

 0   번호           17190 non-null  int64   
 1   개방서비스명       17190 non-null  object   
 2   개방서비스아이디     17190 non-null  object   
 3   개방자치단체코드     17190 non-null  int64    
 4   관리번호         17190 non-null  object   
 5   인허가일자        17190 non-null  int64    
 6   인허가취소일자      0 non-null      float64  
 7   영업상태구분코드     17190 non-null  int64  
 8   영업상태명        17190 non-null  object   
 9   상세영업상태코드     17190 non-null  int64    
 10  상세영업상태명      17190 non-null  object   
 11  폐업일자         10874 non-null  float64  
 12  휴업시작일자       0 non-null      float64  
 13  휴업종료일자       0 non-null      float64  
 14  재개업일자        0 non-null      float64  
 15  소재지전화        15920 non-null  object   
 16  소재지면적        17146 non-null  object   
 17  소재지우편번호      16708 non-null  float64  
 18  소재지전체주소      17169 non-null  object   
 19  도로명전체주소      9508 non-null   object   
 20  도로명우편번호      9230 non-null   float64  
 21  사업장명         17190 non-null  object   
 22  최종수정시점       17190 non-null  int64        
 23  데이터갱신구분      17190 non-null  object   
 24  데이터갱신일자      17190 non-null  object   
 25  업태구분명        17190 non-null  object   
 26  좌표정보(x)      15982 non-null  float64  
 27  좌표정보(y)      15982 non-null  float64  
 28  위생업태명        17190 non-null  object   
 29  건물지상층수       12918 non-null  float64  
 30  건물지하층수       11732 non-null  float64  
 31  사용시작지상층      11407 non-null  float64  
 32  사용끝지상층       8603 non-null   float64  
 33  사용시작지하층      9453 non-null   float64  
 34  사용끝지하층       6529 non-null   float64  
 35  한실수          9793 non-null   float64  
 36  양실수          9794 non-null   float64  
 37  욕실수          12461 non-null  float64  
 38  발한실여부        17111 non-null  object   
 39  의자수          9802 non-null   float64  
 40  조건부허가신고사유    27 non-null     object   
 41  조건부허가시작일자    19 non-null     float64  
 42  조건부허가종료일자    20 non-null     float64  
 43  건물소유구분명      4541 non-null   object   
 44  세탁기수         6619 non-null   float64  
 45  여성종사자수       1800 non-null   float64  
 46  남성종사자수       1800 non-null   float64  
 47  회수건조수        5919 non-null   float64  
 48  침대수          5862 non-null   float64  
 49  다중이용업소여부     17190 non-null  object   
 50  Unnamed: 50  0 non-null      float64  

In [7]:
# 1차 컬럼 삭제
df1 = df.drop(['개방자치단체코드', # 대전지역 데이터라서 삭제
               '인허가일자', # 불필요한 데이터
               '인허가취소일자', # 불필요한 데이터
               '영업상태명', # 영업상태구분코드로 확인 1이면 정상영업 3이면 폐업
               '상세영업상태코드', # 영업상태구분코드로 확인 1이면 정상영업 3이면 폐업
               '상세영업상태명', # 영업상태구분코드로 확인 1이면 정상영업 3이면 폐업
               '폐업일자', # 불필요한 데이터
               '휴업시작일자', # 불필요한 데이터
               '휴업종료일자', # 불필요한 데이터
               '재개업일자', # 불필요한 데이터
               '소재지전화', # 불필요한 데이터
               '소재지면적', # 불필요한 데이터
               '최종수정시점', # 불필요한 데이터
               '데이터갱신구분', # 불필요한 데이터
               '데이터갱신일자', # 불필요한 데이터
               '업태구분명', # 개방서비스명으로 구분
               '위생업태명', # 개방서비스명으로 구분
               '건물지상층수', # 불필요한 데이터
               '건물지하층수', # 불필요한 데이터
               '사용시작지상층', # 불필요한 데이터
               '사용끝지상층', # 불필요한 데이터
               '사용시작지하층', # 불필요한 데이터
               '사용끝지하층', # 불필요한 데이터
               '한실수', # 불필요한 데이터
               '양실수', # 불필요한 데이터
               '욕실수', # 불필요한 데이터
               '발한실여부', # 불필요한 데이터
               '의자수', # 불필요한 데이터
               '조건부허가신고사유', # 불필요한 데이터
               '조건부허가시작일자', # 불필요한 데이터
               '조건부허가종료일자', # 불필요한 데이터
               '건물소유구분명', # 불필요한 데이터
               '세탁기수', # 불필요한 데이터
               '여성종사자수', # 불필요한 데이터
               '남성종사자수', # 불필요한 데이터
               '회수건조수', # 불필요한 데이터
               '침대수', # 불필요한 데이터
               '다중이용업소여부', # 불필요한 데이터
               'Unnamed: 50' # 불필요한 데이터
               ], axis=1)

In [8]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17190 entries, 0 to 17189
Data columns (total 12 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   번호        17190 non-null  int64  
 1   개방서비스명    17190 non-null  object 
 2   개방서비스아이디  17190 non-null  object 
 3   관리번호      17190 non-null  object 
 4   영업상태구분코드  17190 non-null  int64  
 5   소재지우편번호   16708 non-null  float64
 6   소재지전체주소   17169 non-null  object 
 7   도로명전체주소   9508 non-null   object 
 8   도로명우편번호   9230 non-null   float64
 9   사업장명      17190 non-null  object 
 10  좌표정보(x)   15982 non-null  float64
 11  좌표정보(y)   15982 non-null  float64
dtypes: float64(4), int64(2), object(6)
memory usage: 1.6+ MB


In [23]:
df1.tail()

번호 개방서비스명    개방서비스아이디                    관리번호  영업상태구분코드   소재지우편번호  \
17185  17186   목욕장업  11_44_01_P  4570000-202-2001-00001         3  323811.0   
17186  17187   목욕장업  11_44_01_P  3340000-202-2018-00001         3  604040.0   
17187  17188   목욕장업  11_44_01_P  4850000-202-2018-00002         3  517802.0   
17188  17189   목욕장업  11_44_01_P  5650000-202-2018-00001         3  330993.0   
17189  17190   목욕장업  11_44_01_P  4390000-202-2021-00001         3  380090.0   

                                     소재지전체주소  \
17185                충청남도 부여군 규암면 규암리 40-8번지   
17186   부산광역시 사하구 장림동 1151 사하장림역스마트W아파트 B01호   
17187                 전라남도 담양군 담양읍 객사리 264-9   
17188  충청남도 천안시 동남구 신부동 401-40 외1필지, 2동 지하1층   
17189                      충청북도 충주시 용산동 2116   

                                                 도로명전체주소  도로명우편번호       사업장명  \
17185                                                NaN      NaN   리버사이드목욕장   
17186  부산광역시 사하구 장림번영로104번길 10, 지하1층 B01호 (장림동, 사하장림역...  49465.0      W 사우나   
17187                            전라남도 담양군 담양읍 추성로 1335-1  57340.0  아모레 여성사우나   
17188            충청남도 천안시 동남구 천안천10길 9-15, 2동 지하1층 (신부동)  31115.0     제이피사우나   
17189                            충청북도 충주시 남산로 34-1 (용산동)  27401.0      미인만들기   

             좌표정보(x)        좌표정보(y)  
17185  189776.913686  308445.385341  
17186  380184.728683  177449.877784  
17187  198495.864694  202290.518322  
17188  214048.598187  369407.120829  
17189  283281.845147  384808.815293

폐업 상태인 목욕장 삭제, `영업상태구분코드`가 3이면 폐업

In [24]:
df_remove = df1['영업상태구분코드'].isin([3])
df2 = df1[~df_remove]
df2.tail()

번호 개방서비스명    개방서비스아이디                    관리번호  영업상태구분코드   소재지우편번호  \
6311  6312   목욕장업  11_44_01_P  3420000-202-2021-00001         1  701827.0   
6312  6313   목욕장업  11_44_01_P  3220000-202-2021-00002         1  135892.0   
6313  6314   목욕장업  11_44_01_P  4480000-202-2021-00001         1  395904.0   
6314  6315   목욕장업  11_44_01_P  4670000-202-2021-00001         1  573866.0   
6315  6316   목욕장업  11_44_01_P  5130000-202-2021-00001         1  712130.0   

                                        소재지전체주소  \
6311  대구광역시 동구 신천동 326-1 대구 메리어트 호텔 및 서비스드 레지던스   
6312                    서울특별시 강남구 신사동 585-1 네오스   
6313                      충청북도 단양군 매포읍 평동리 1227   
6314                        전라북도 군산시 나운동 486-12   
6315                       경상북도 경산시 대평동 77-3 3층   

                                                도로명전체주소  도로명우편번호      사업장명  \
6311  대구광역시 동구 동부로26길 6, 대구 메리어트 호텔 및 서비스드 레지던스 5층 (...  41243.0  웰니스클럽사우나   
6312            서울특별시 강남구 논현로 831, 네오스 지하1층, 지하2층 (신사동)   6032.0        마루   
6313                               충청북도 단양군 매포읍 평동로 122  27005.0  매포 복지목욕탕   
6314                      전라북도 군산시 나운2길 5, 1,2,3층 (나운동)  54139.0    군산더사우나   
6315                        경상북도 경산시 강변동로 378, 3층 (대평동)  38500.0   내외리버스퀘어   

            좌표정보(x)        좌표정보(y)  
6311  346922.110876  265123.315824  
6312  202378.390574  446705.197149  
6313  315613.611425  393144.059829  
6314  172936.805144  274065.522127  
6315  356067.918857  261584.638320

In [32]:
# 2차 컬럼 삭제
df3 = df2.drop(['개방서비스명', # 불필요한 정보
                '개방서비스아이디', # 불필요한 정보
                '관리번호', # 불필요한 정보
                '영업상태구분코드' # 불필요한 정보
                ], axis=1)
df3.head()

번호   소재지우편번호                   소재지전체주소                  도로명전체주소  도로명우편번호  \
0   1  718832.0   경상북도 칠곡군 석적읍 반계리 71-4번지   경상북도 칠곡군 석적읍 호국로 582-2  39848.0   
1   2  718852.0   경상북도 칠곡군 기산면 행정리 47-2번지    경상북도 칠곡군 기산면 선노로 1082  39915.0   
2   3  718821.0     경상북도 칠곡군 지천면 송정리 45번지  경상북도 칠곡군 지천면 송정도당길 67-9  39867.0   
3   4  718813.0  경상북도 칠곡군 약목면 복성리 859-1번지    경상북도 칠곡군 약목면 약목로 87-8  39825.0   
4   5  718811.0   경상북도 칠곡군 약목면 남계리 73-1번지      경상북도 칠곡군 약목면 약목로 49  39827.0   

         사업장명        좌표정보(x)        좌표정보(y)  
0    반계숯가마찜질방  329518.461747  282201.682569  
1  무시로장수마을찜질방  325101.041208  275060.260588  
2    솔밭에온가마찜질  332456.755220  275605.443845  
3       복성목욕탕  322935.849074  282378.690977  
4       약목목욕탕  323189.818600  282060.794671

In [34]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 6316 entries, 0 to 6315
Data columns (total 8 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   번호       6316 non-null   int64  
 1   소재지우편번호  6177 non-null   float64
 2   소재지전체주소  6302 non-null   object 
 3   도로명전체주소  6263 non-null   object 
 4   도로명우편번호  6208 non-null   float64
 5   사업장명     6316 non-null   object 
 6   좌표정보(x)  6268 non-null   float64
 7   좌표정보(y)  6268 non-null   float64
dtypes: float64(4), int64(1), object(3)
memory usage: 444.1+ KB


CSV 파일로 저장

In [33]:
df3.to_csv(PATH + '목욕장업_전처리.csv', index=False, encoding='cp949')